binota/bangumi-homepage-calendar.user.js

```diff
@@ -44,14 +44,24 @@ const DEFAULT_CONFIG = {
 var formhash = '';
   
 var $ = function() { return document.querySelector(arguments[0]); };
-var get = function() {
-  var url = arguments[0];
-  var sync = (typeof arguments[1] === 'undefined') ? true : arguments[1];
-  var req = new XMLHttpRequest();
-  req.open('GET', url, false);
-  req.send(null);
- 
-  if(req.status === 200) return req.responseText;
+var get = function (url, sync) {
+  return new Promise((resolve, reject) => {
+    var sync = (typeof sync === 'undefined') ? true : sync;
+    var req = new XMLHttpRequest();
+    req.onload = () => {
+      if (req.status === 200)
+        return resolve(req.responseText);
+      else
+        return reject();  
+    };
+    req.open('GET', url, !sync);
```

Comment:
It seems to be a little confused. We passed a sync argument, but the bahavior in the function seems always use syncronouse request.
Can you just rewrite whole `get` function, and use `fetch()` as a replacement for `Promise + XMLHttpRequest`?